code/datums/diseases/black_goo.dm

```diff
@@ -16,117 +24,117 @@
 	longevity = 500 //the virus tend to die before the dead is turn into zombie this should fix it.
 	stage_prob = 0//no randomness
 
-	/// whether we're currently transforming the host into a zombie.
-	var/zombie_transforming = 0
-	/// tells a dead infectee their stage, so they can know when-abouts they'll revive
+	/// boolean value to determine if the mob is currently transforming into a zombie.
+	var/zombie_is_transforming = FALSE
+
+	/// variable to keep track of the stage level, used to prevent the stage message from being displayed more than once for any given stage.
 	var/stage_counter = 0
 
 //new variables to handle infection progression inside a stage.
 
-	/// variable that contain accumulated virus progression for an host.
+	/// variable that contains accumulated virus progression for a host. Iterates to a value above 360 and is then reset.
 	var/stage_level = 0
-	/// variable that handle passive increase of the virus of an host.
-	var/infection_rate = 1
 
-	///the number of stage level needed to pass another stage.
-	var/stage_level_check = 360
+	/// variable that handles passive increase of the virus of a host.
+	var/infection_rate = SLOW_INFECTION_RATE
 
-	/// cooldown between each check to see if we display a symptome idea is to get 60s between symptome atleast.
-	var/message_cooldown_time = 1 MINUTES
+	/// cooldown for the living mob's symptom messages
 	COOLDOWN_DECLARE(goo_message_cooldown)
 
 /datum/disease/black_goo/stage_act()
 	..()
-	if(!ishuman(affected_mob)) return
-	var/mob/living/carbon/human/H = affected_mob
+	if(!ishuman_strict(affected_mob))
+		return
+	var/mob/living/carbon/human/infected_mob = affected_mob
 
-	// check if your already a zombie or in the process of being transform into one...
-	if(iszombie(H))
+	if(iszombie(infected_mob))
 		return
 
-	// check if dead
-	if(H.stat == DEAD)
-		infection_rate = 4
+	// infection rate is faster for dead mobs
+	if(infected_mob.stat == DEAD)
+		infection_rate = FAST_INFECTION_RATE
```

Comment:
This seems to be an unexplained change - why is the infection rate 7 when it was 4? Fine to do so, but mention it.

Reply:
Yeah, the problem was that before mobs would experience symptoms even when dead, and when you experienced a symptom depending on which symptom it was it added to the stage_level. So since I removed dead mobs being able to have symptoms I just made the infection rate slightly faster to compensate for that.